.github/scripts/ghcr-prune.py

```diff
@@ -0,0 +1,171 @@
+import argparse
+import logging
+import requests
+import re
+import json
+from datetime import datetime
+from datetime import timedelta
+
+parser = argparse.ArgumentParser()
+
+parser.add_argument("--token", required=True, help='GitHub token with "repo" scope')
+parser.add_argument("--org", required=True, help="Organization name")
+parser.add_argument("--name", required=True, help="Package name")
+parser.add_argument(
+    "--age", type=int, help="Filter versions by age, removing anything older than"
+)
+parser.add_argument(
+    "--filter", help="Filter which versions are consider for pruning", default=".*"
+)
+parser.add_argument("--untagged", action="store_true", help="Prune untagged versions")
+parser.add_argument(
+    "--dry-run", action="store_true", help="Does not actually delete anything"
+)
+
+logging_group = parser.add_argument_group("logging")
+logging_group.add_argument(
+    "--log-level", choices=("DEBUG", "INFO", "WARNING", "ERROR"), default="INFO"
+)
+
+kwargs = vars(parser.parse_args())
+
+logging.basicConfig(level=kwargs["log_level"])
+
+logger = logging.getLogger("ghcr-prune")
+
+
+class GitHubPaginate:
```

Comment:
Can you document this a little - what is it intended to do?

Reply:
Adds description to tool and docstring for class. See the ghcr-prune workflow for usage.